web-registry/src/components/organisms/BuyCreditsModal/BuyCreditsModal.tsx

```diff
@@ -195,26 +194,24 @@ const BuyCreditsModal: React.FC<BuyCreditsModalProps> = ({
           validate={validationHandler}
           onSubmit={async (values, { setSubmitting }) => {
             setSubmitting(true);
-            if (location.pathname.includes('projects')) {
-              const track_data: any = {
```

Comment:
I guess something like this
```
interface TrackBuyModal {   // instead of TrackBuy2 ?
	price?: string;
	batchDenom?: string;
	projectName?: string;
	creditClassName?: string;
	amountTradable?: string;
	amountRetired?: string;
}
```
so you can do
```
const trackData: TrackBuyModal = {
```

Reply:
@wgwz also `trackData` instead of `track_data` ?

Reply:
i'm not sure what having the type adds here, because it's only one place that we would be using this type, and the `track` function accepts `any` as the second argument. i'm open to adding it but would like to understand the reasons it's a good thing to do

Reply:
1. since `track` accepts `any`,  it will accept any type that we define
2. since we do know how the data is supposed to look like, best to avoid `any` and define a precise type, it's best for type safety and documentation. Since it's not reused anywhere else, you could also just do:
```
const trackData: {
  price?: string;
  ...
} = {
  // actual data
};

Reply:
useful read: https://thoughtbot.com/blog/typescript-stop-using-any-there-s-a-type-for-that

Reply:
also see: https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#any

Reply:
thanks for the reading, that makes more sense now, why we should want to add a type here. i've used the inline type approach you mentioned @blushi : e985c1d4

Reply:
sorry, about the `track_data` vs `trackData`, it's a force of habit from my days working with python 😄